frontend/src/components/MetacodeSelect.js

```diff
@@ -0,0 +1,168 @@
+/* global $ */
+
+import React, { Component, PropTypes } from 'react'
+
+const ENTER_KEY = 13
+const LEFT_ARROW = 37
+const UP_ARROW = 38
+const RIGHT_ARROW = 39
+const DOWN_ARROW = 40
+
+const Metacode = (props) => {
+  const { m, onClick, underCursor } = props
+  
+  return (
+    <li onClick={() => onClick(m.id) } className={ underCursor ? 'keySelect' : '' }>
+      <img src={ m.get('icon') } />
+      <span>{ m.get('name') }</span>
+    </li>
+  )
+}
```

Comment:
Do you think we might ever re-use this component? We could pull it out into its own file if you think we could

That might involve passing some of the HTML as props to make it more useful (but we could add that abstraction when we get there, and leave <li> as the default)